### DISCUSSION TITLE
Custom server on mobile apps?

### DISCUSSION THREAD

**Discussion by ghost** [padloc/padloc#442] [Q&A] [answered]
Hello. I'm interested in running self-hosted padloc, and I need mobile clients (especially iOS) to support it. Is this being planned to implement?

**Comment by zmactep** [comment]
Hi, it would be great, as you cannot set the PWA as a password provider in OS. So passwords autocomplete doesn't work with custom servers.

**Comment by MaKleSoft** [comment, marked as answer]
Hi there! For security reasons, we do not currently support pointing the official apps to a custom server. However, are working on improving the tooling and documentation around building your own custom version of all apps and extensions! Additionally, we are considering releasing a "community edition" of all apps, which can be configured to be used with a custom server.

**Comment by BrunoBernardino** [comment]
We've published guides for self-hosting apps/extensions here: https://github.com/padloc/padloc/tree/main/docs/examples/hosting/app

**Comment by c4lliope** [comment]
Hello all; I'm sorry to be the one to make this recommendation, though I can claim no patience for a program which disregards the needs of their users as though they're meaningless. Anyone seeing this kind of issue (basically, all besides the paying customers) should begin using https://www.passbolt.com/ - their apps and browser addons ask for the custom URL on the first screen, and since July 2022 I've been using them with no issue - simly make sure you keep a copy of the GPG key you're assigned on signup so you can log in from other machines.

I had high hopes for Padloc upon seeing such slick design and clear security audits displayed in their branding. I had even been able to deploy the program on my homelab's Dokku alongside all my homemade code. After a couple days of hacking on the deployment I'd been sure I had a good solution. Then as soon as I reached for the browser extension and android apps I realized I had a serious problem.

11 months ago, @MaKleSoft had said:

> Hi there! For security reasons, we do not currently support pointing the official apps to a custom server. However, are working on improving the tooling and documentation around building your own custom version of all apps and extensions! Additionally, we are considering releasing a "community edition" of all apps, which can be configured to be used with a custom server.

The "security" reason here is a red herring - custom domains are likely more secure than a single high-profile corporate domain handling all accounts. Padloc's programmers are saying, "you can run this yourself - if you can manage as large a presence on the app and browser stores as we do." This precludes any usage by the open-source coders who would likely help to make upgrades to the codebase after using it for personal or company use.

Padloc is disregarding user needs because they're picky regarding who their users are.
I do hope they make good on their promise of a "community edition" and once this happens I may rescind this.
For now, I'll be going back to https://www.passbolt.com/ and I recommend the same for many of you.
